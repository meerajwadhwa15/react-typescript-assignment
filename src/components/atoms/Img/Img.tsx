import React, { FunctionComponent } from 'react'

export type Props = {
  src: string
  alt: string
  width?: string
  height?: string
}

const Img: FunctionComponent<Props> = (props) => <img {...props} />

export default Img
