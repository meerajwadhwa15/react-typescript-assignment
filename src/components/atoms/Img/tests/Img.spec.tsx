import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Img, { Props } from '../Img'

const props: Props = {
  src: 'https://images.punkapi.com/v2/keg.png',
  alt: 'credit card',
  width: '200',
  height: '200',
}

describe('Image Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(<Img {...props} />)
  })
})

describe('Image Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(<Img {...props} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
