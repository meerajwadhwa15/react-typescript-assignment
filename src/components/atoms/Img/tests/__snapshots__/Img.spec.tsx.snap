// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshot testing Matches the snapshot 1`] = `
<img
  alt="credit card"
  height="200"
  src="https://images.punkapi.com/v2/keg.png"
  width="200"
/>
`;
